components/data_prefetch/data_prefetch.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PQueue from 'p-queue';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Constants} from 'utils/constants';
+import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
+
+const queue = new PQueue({concurrency: 2});
+
+type Props = {
+    currentChannelId: string;
+    prefetchQueueObj: Record<string, string[]>;
+    prefetchRequestStatus: Record<string, any>;
+    unreadChannels: Channel[];
+    actions: {
+        prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<any>;
+    };
+}
+
+export default class DataPrefetch extends React.PureComponent<Props, {}> {
+    private prefetchTimeout?: number;
+
+    async componentDidUpdate(prevProps: Props) {
+        const {currentChannelId, prefetchQueueObj} = this.props;
+        if (!prevProps.currentChannelId && currentChannelId) {
+            queue.add(async () => this.prefetchPosts(currentChannelId));
+            await loadProfilesForSidebar();
+            this.prefetchData();
+        } else if (prevProps.prefetchQueueObj !== prefetchQueueObj) {
+            clearTimeout(this.prefetchTimeout);
+            queue.clear();
+            this.prefetchTimeout = window.setTimeout(() => {
+                this.prefetchData();
+            }, 0);
+        }
+    }
+
+    public componentWillUnmount(): void {
+        clearTimeout(this.prefetchTimeout);
+    }
+
+    public prefetchPosts = (channelId: string) => {
+        let delay;
+        const channel = this.props.unreadChannels.find((unreadChannel) => channelId === unreadChannel.id);
+        if (channel && (channel.type === Constants.PRIVATE_CHANNEL || channel.type === Constants.OPEN_CHANNEL)) {
```

Comment:
Why only add a delay for public/private channels?

Reply:
Add a jitter(0-1sec) for delaying post requests in case of a new message in open/private channels. This is to prevent a case when all clients request messages when new post is made in a channel with thousands of users.

Some of the your questions might be answered if you just read the summary again. 

DM/GM's do not have enough users to cause an issue on the server

Reply:
Don't we have something to track whether or not a channel is in sync though? We shouldn't need to fire a prefetch for new messages when the channel is still in sync.

Also, could you add a comment to explain why we need a delay?

Reply:
>  We shouldn't need to fire a prefetch for new messages when the channel is still in sync.

yes, we don't.

This is the case when the channel did not have any unread so there was never a prefetch before. On new message say in bugs channel we don't want all users making the call for unreads at once hence the jitter.